internal/version/version.go

```diff
@@ -4,14 +4,16 @@ import (
 	"fmt"
 	"log"
 	"runtime"
-	"runtime/debug"
-	"strconv"
 	"time"
 )
 
+const gitTreeStateClean = "clean"
+
 var (
-	version   = ""                     // Injected with a linker flag
-	buildDate = "1970-01-01T00:00:00Z" // Injected with a linker flag
+	version      = ""                     // Injected with a linker flag
+	buildDate    = "1970-01-01T00:00:00Z" // Injected with a linker flag
+	gitCommit    = ""                     // Injected with a linker flag
+	gitTreeState = ""                     // Injected with a linker flag
```

Comment:
These used to be inferred from build info at runtime, but that build info doesn't include any git details since the `.git/` dir is not present inside the container at build time. i.e. That's the source of the bug.

We'll let these be set through linker args instead.